helix-tui/src/buffer.rs

```diff
@@ -409,6 +409,8 @@ impl Buffer {
         let max_x_offset = min(self.area.right() as usize, width.saturating_add(x as usize));
 
         for s in string.graphemes(true) {
+            let s = if s == "\n" { " " } else { s };
```

Comment:
This modifies a pretty crucial rendering function used in various places. I think instead we want to pre-process the status message before rendering